operator/v1alpha1/types_etcdbackup.go

```diff
@@ -0,0 +1,101 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:resource:scope=Cluster
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// # EtcdBackup provides configuration options and status for a one-time backup attempt of the etcd cluster
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type EtcdBackup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec EtcdBackupSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Status EtcdBackupStatus `json:"status"`
+}
+
+type EtcdBackupSpec struct {
+	// PVCName specifies the name of the PersistentVolumeClaim (PVC) which binds a PersistentVolume where the
+	// etcd backup file would be saved
+	// The PVC itself must always be created in the "openshift-etcd" namespace
+	// If the PVC is left unspecified "" then the platform will choose a reasonable default location to save the backup.
+	// In the future this would be backups saved across the control-plane master nodes.
+	// +kubebuilder:validation:Optional
+	// +optional
```

Comment:
Sorry, caught this in the unit test. The config.openshift.io API must accept an empty PVCName, but this object needs one so that the pods are assigned well, right?

Reply:
Depends on where we want to handle the no-option local backup signal.

My rough idea is to handle the `pvcName: ""` at the `operator.openshift.io EtcdBackup` object for each request.
That would signal to the etcdbackup controller to pick a PVC from its own set of pre-created PVCs that are bound to the local PVs (one for each node).
How the node selectors (or rather picking from the PVs assigned to each node) is decided based on the history of previous backups and existing spread.

But essentially in this model, the no-option `pvcName: ""` from the `config.openshift.io Backup` object is just carried over to the `operator.openshift.io EtcdBackup` object, which is interpreted as the default local backup option by the lower level controller.

So this object also needs to accept an empty `pvcName` in that case.